src/dashboard/src/main/management/commands/update_elasticsearch_mappings.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+"""Update Elasticsearch mappings via Put Mapping.
+
+Updates the Elasticsearch mappings for the aips and aipfiles indices for
+Archivematica 1.12 to enable sorting on all fields displayed in the new
+Archival Storage DataTable.
+
+Execution example:
+
+./manage.py update_elasticsearch_mappings
+"""
+from __future__ import absolute_import, print_function
+
+import sys
+
+from django.conf import settings
+from elasticsearch import ElasticsearchException
+
+from main.management.commands import DashboardCommand
+import elasticSearchFunctions as es
+
+
+class Command(DashboardCommand):
+
+    help = __doc__
+
+    def handle(self, *args, **options):
+        # Check that AIPs index is enabled
+        if es.AIPS_INDEX not in settings.SEARCH_ENABLED:
+            self.error(
+                "The AIPs indexes are not enabled. Please, make sure to "
+                "set the *_SEARCH_ENABLED environment variables to `true` "
+                "to enable the AIPs and Transfers indexes, or to `aips` "
+                "to only enable the AIPs indexes."
+            )
+            sys.exit(1)
+
+        es.setup_reading_from_conf(settings)
+        es_client = es.get_client()
+
+        try:
+            es_client.info()
+        except ElasticsearchException:
```

Comment:
Unlike you would expect, we check the indexes existence while setting up the client in ... https://github.com/artefactual/archivematica/blob/qa/1.x/src/archivematicaCommon/lib/elasticSearchFunctions.py#L202.

So you may want to include the setup and client retrieval instead of the info call.

Reply:
Ups, clicked single comment but I'm on it @tw4l ;)